http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -461,4 +545,124 @@ private void addScopes(String scopes, Set<String> allScopes) {
+                Key key = new HmacKey(deployment.getResourceCredentials().get("secret").toString().getBytes(StandardCharsets.UTF_8));   //the client secret is a shared secret between the server and the client
+                jsonWebSignature.setDoKeyValidation(false);     //skips validation so that size of the secret does not matter
+                jsonWebSignature.setKey(key);
+            } else { //signed with asymmetric key
+                KeyPair keyPair = getkeyPair(deployment);
+                jsonWebSignature.setKey(keyPair.getPrivate());
+                jsonWebSignature.setAlgorithmHeaderValue(deployment.getRequestObjectSignatureAlgorithm());
+            }
+            jsonWebSignature.sign();
+            return jsonWebSignature;
+        }
+    }
+
+    private JsonWebEncryption encryptRequest(JsonWebSignature signedRequest) throws JoseException, IOException {
+        if (!deployment.getRequestObjectEncryptionAlgValuesSupported().contains(deployment.getRequestObjectEncryptAlgorithm())) {
+            throw log.invalidRequestObjectEncryptionAlgorithm();
+        } else if (!deployment.getRequestObjectContentEncryptionMethodsSupported().contains(deployment.getRequestObjectContentEncryptionMethod())) {
+            throw log.invalidRequestObjectContentEncryptionAlgorithm();
+        } else {
+            JsonWebEncryption jsonEncryption = new JsonWebEncryption();
+            jsonEncryption.setPayload(signedRequest.getCompactSerialization());
+            jsonEncryption.setAlgorithmConstraints(new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.PERMIT, deployment.getRequestObjectEncryptAlgorithm(), deployment.getRequestObjectContentEncryptionMethod()));
+            jsonEncryption.setAlgorithmHeaderValue(deployment.getRequestObjectEncryptAlgorithm());
+            jsonEncryption.setEncryptionMethodHeaderParameter(deployment.getRequestObjectContentEncryptionMethod());
+            PublicKey encPublicKey = deployment.getEncryptionPublicKeyLocator().getPublicKey("enc", deployment);
+            jsonEncryption.setKey(encPublicKey);
+            return jsonEncryption;
+        }
+    }
+
+    private String getRequestUri(String request) throws OidcException {
+        HttpPost parRequest = new HttpPost(deployment.getPushedAuthorizationRequestEndpoint());
+        List<NameValuePair> formParams = new ArrayList<NameValuePair>();
+        formParams.add(new BasicNameValuePair(REQUEST, request));
+        ClientCredentialsProviderUtils.setClientCredentials(deployment, parRequest, formParams);
+        parRequest.addHeader("Content-type", "application/x-www-form-urlencoded");
+
+        UrlEncodedFormEntity form = new UrlEncodedFormEntity(formParams, StandardCharsets.UTF_8);
+        parRequest.setEntity(form);
+
+        HttpResponse response;
+        try {
+            response = deployment.getClient().execute(parRequest);
+        } catch (Exception e) {
+            throw log.failedToConnectToOidcProvider(e);
```

Comment:
Can we make this log message more specific? e.g., it would be good to indicate there was an issue with sending the request to the OpenID provider's pushed authorization request endpoint.

Reply:
Changes upto this comment has been addressed in https://github.com/wildfly-security/wildfly-elytron/pull/1984/commits/a52d5a2db27f8d567fa3c43e1c055d98f931bc2b